### PR TITLE
Spring Security annotations  on subclasses support intercepting parent class methods.

### PR DESCRIPTION
**Pull request by kse-music**
Closes [gh-13783](https://github.com/spring-projects/spring-security/issues/13783)

### CODE REVIEW

File: core/src/test/java/org/springframework/security/authorization/method/Jsr250AuthorizationManagerTests.java [hunk 1]
```diff
@@ -225,6 +225,48 @@ public void checkInheritedAnnotationsWhenConflictingThenAnnotationConfigurationE
...
+	public void checkDenyAllWhenMethodsFromInheritThenApplies() throws Exception {
+		MockMethodInvocation methodInvocation = new MockMethodInvocation(new DenyAllClass(), DenyAllClass.class,
+				"securedUser");
+		Jsr250AuthorizationManager manager = new Jsr250AuthorizationManager();
+		AuthorizationDecision decision = manager.check(TestAuthentication::authenticatedUser, methodInvocation);
+		assertThat(decision.isGranted()).isFalse();
+	}
+
+	@RolesAllowed("USER")
+	public static class RolesAllowedClass extends SecuredAuthorizationManagerTests.ParentClass {
```
Comment (jzheaux) [inline core/src/test/java/org/springframework/security/authorization/method/Jsr250AuthorizationManagerTests.java:256]: Please copy the definition for `ParentClass` in this class. That way, the test classes don't refer to each other.

File: core/src/test/java/org/springframework/security/authorization/method/PostAuthorizeAuthorizationManagerTests.java [hunk 2]
```diff
@@ -167,6 +167,21 @@ public void checkInheritedAnnotationsWhenConflictingThenAnnotationConfigurationE
...
+		MockMethodInvocation methodInvocation = new MockMethodInvocation(new PostAuthorizeClass(),
+				PostAuthorizeClass.class, "securedUser");
+		MethodInvocationResult result = new MethodInvocationResult(methodInvocation, null);
+		PostAuthorizeAuthorizationManager manager = new PostAuthorizeAuthorizationManager();
+		AuthorizationDecision decision = manager.check(TestAuthentication::authenticatedUser, result);
+		assertThat(decision.isGranted()).isTrue();
+	}
+
+	@PostAuthorize("hasRole('USER')")
+	public static class PostAuthorizeClass extends SecuredAuthorizationManagerTests.ParentClass {
```
Comment (jzheaux) [inline core/src/test/java/org/springframework/security/authorization/method/PostAuthorizeAuthorizationManagerTests.java:181]: Please copy the definition for `ParentClass` in this class. That way, the test classes don't refer to each other.

Please also do this with any other references to the `ParentClass` defined in another test class.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @kse-music! This is looking good; I've left some feedback inline.

Also, when you submit the requested changes, will you please squash your commits and have the commit message contain a title and the ticket like so:

```
Have Method Security Start at Target Class

Closes gh-13783
```

Feel free to have a different title, the above is just the requested format.

**Comment by jzheaux** [comment]
Also, I'd recommend rebasing so that you don't need to the two extra formatting changes as well.

**Comment by kse-music** [comment]
hi @jzheaux ,I have opened a new pr #14516, and completed all your suggestions

**Pull request closed without merging**